src/main/java/dev/j3fftw/litexpansion/machine/WaterSprinkler.java

```diff
@@ -0,0 +1,152 @@
+        "&cNo water found",
+        "",
+        "&cPlease place water under the sprinkler!"
+    );
+    private static final CustomItem waterFoundItem = new CustomItem(Material.WATER_BUCKET,
+        "&bWater detected"
+    );
+
+    public WaterSprinkler() {
+        super(Items.LITEXPANSION, Items.WATER_SPRINKER, RecipeType.ENHANCED_CRAFTING_TABLE,
+            new ItemStack[] {
+                Items.REFINED_IRON, SlimefunItems.ELECTRIC_MOTOR, Items.REFINED_IRON,
+                new ItemStack(Material.BUCKET), Items.MACHINE_BLOCK, new ItemStack(Material.BUCKET),
+                Items.REFINED_IRON, SlimefunItems.MEDIUM_CAPACITOR, Items.REFINED_IRON
+            });
+
+        createPreset(this, Items.WATER_SPRINKER.getImmutableMeta().getDisplayName().orElse("&bWater Sprinkler"),
+            blockMenuPreset -> {
+                for (int i = 0; i < 9; i++)
+                    blockMenuPreset.addItem(i, ChestMenuUtils.getBackground(), ChestMenuUtils.getEmptyClickHandler());
+
+                blockMenuPreset.addItem(PROGRESS_SLOT, noWaterItem);
+            });
+
+        crops.add(Material.WHEAT);
+        crops.add(Material.POTATOES);
+        crops.add(Material.CARROTS);
+        crops.add(Material.NETHER_WART);
+        crops.add(Material.BEETROOTS);
+        crops.add(Material.COCOA);
+        crops.add(Material.SUGAR_CANE);
+
+        if (SlimefunPlugin.getMinecraftVersion().isAtLeast(MinecraftVersion.MINECRAFT_1_14)) {
+            crops.add(Material.SWEET_BERRY_BUSH);
+        }
+    }
+
+    public int getEnergyConsumption() {
+        return ENERGY_CONSUMPTION;
+    }
+
+    @Override
+    public int getCapacity() {
+        return CAPACITY;
+    }
+
+    public int getRadius() {
+        return RADIUS;
+    }
+
+    public int[] getInputSlots() {
+        return new int[0];
+    }
+
+    public int[] getOutputSlots() {
+        return new int[0];
+    }
+
+    @Override
+    protected void tick(Block b) {
+        BlockMenu inv = BlockStorage.getInventory(b);
+
+        if (b.getRelative(BlockFace.DOWN).getType() == Material.WATER) {
+            inv.replaceExistingItem(PROGRESS_SLOT, waterFoundItem);
+        } else {
+            inv.replaceExistingItem(PROGRESS_SLOT, noWaterItem);
+            return;
+        }
+
+        if (getCharge(b.getLocation()) >= getEnergyConsumption()) {
+            for (int x = -getRadius(); x <= getRadius(); x++) {
+                for (int z = -getRadius(); z <= getRadius(); z++) {
+                    Block block = b.getRelative(x, 0, z);
+
+                    if (Constants.SPRINKLER_PARTICLES)
+                    block.getWorld().spawnParticle(Particle.WATER_SPLASH, block.getLocation().add(0.5D, 0.5D, 0.5D), 4, 0.1F, 0.1F, 0.1F);
+
+                    if (crops.contains(block.getType())) {
+                        grow(block);
+                    }
+                }
+            }
+        }
+    }
+
+    private void grow(Block crop) {
+
+        double random = Math.random();
```

Comment:
This random is not secure. You should instead be doing `ThreadLocalRandom.current().nextDouble()`